File: iOS/swiftui-journey-module/JourneyModuleSample/JourneyModuleSample/ViewModels/LogOutViewModel.swift
// 
//  LogOutViewModel.swift
//  JourneyModuleSample
//
//  Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
//
//  This software may be modified and distributed under the terms
//  of the MIT license. See the LICENSE file for details.
//


import SwiftUI
import Combine
import PingOidc
import PingJourney

/// A view model responsible for managing the logout functionality.
/// - Handles the logout process for the user and updates the state for UI display.
@MainActor
class LogOutViewModel: ObservableObject {
    /// Performs the user logout process using the DaVinci SDK.
    /// - Executes the `logout()` method from the DaVinci or Journey user object asynchronously.
    func logout() async {
        let journeyUser = await journey.journeyUser()
        await journeyUser?.logout()
    }
}

File: android/kotlin-journey/journey/src/main/java/com/pingidentity/samples/journeyapp/LogoutViewModel.kt
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package com.pingidentity.samples.journeyapp

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.pingidentity.journey.user
import com.pingidentity.samples.journeyapp.env.journey
import kotlinx.coroutines.launch

class LogoutViewModel : ViewModel() {
    fun logout(onCompleted: () -> Unit) {
        viewModelScope.launch {
            //If you are using Journey, you can use the Journey user object to logout
            journey.user()?.logout()
        }
        onCompleted()
    }
}